sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala

```diff
@@ -126,6 +126,35 @@ private[sql] trait SQLTestUtils
   }
 
   /**
+   * Generates the requested number of temporary paths without creating the
+   * actual files/directories and passes these to the provided function 'f'.
+   * Deletes any files/directories left at these paths after 'f' returns.
+   *
+   * @param numPaths Number of paths to create
+   * @param f Function to invoke with the created paths
+   */
+  protected def withTempPaths(numPaths: Int)(f: List[File] => Unit) {
+    def addPaths(numPaths: Int, paths: List[File]): List[File] = {
+
+      if (numPaths <= 0) {
+        paths
+      } else {
+        val path = Utils.createTempDir()
+        path.delete()
+        addPaths(numPaths - 1, path :: paths)
+      }
+    }
+
+    val paths = addPaths(numPaths, List.empty)
+
+    try {
+      f(paths)
+    } finally {
+      paths.foreach(path => Utils.deleteRecursively(path))
+    }
+  }
```

Comment:
I personally think It's strange to use a recursive function, does the following modified function work for your test?
```

  protected def withTempDirs(num: Int)(body: Seq[File] => Unit) {
    val files = mutable.Buffer[File]()
    for (i <- 0 until num) {
      files += Utils.createTempDir().getCanonicalFile
    }
    try {
      body(files)
    } finally {
      files.foreach(Utils.deleteRecursively)
    }
  }
```